lib/jekyll/commands/serve.rb

```diff
@@ -99,9 +103,9 @@ def webrick_opts(opts)
         # Recreate NondisclosureName under utf-8 circumstance
 
         private
-        def file_handler_opts
+        def file_handler_opts(opts)
           WEBrick::Config::FileHandler.merge({
-            :FancyIndexing     => true,
+            :FancyIndexing     => opts['show_dir_listing'] ? true : false,
```

Comment:
@hgoodman FancyIndexing by default was a conscious disparity between production and development to enforce WEBrick only being dev quality (as used by us) and to discourage the many users we (or maybe just me?) using it in production.  People dislike directory indexing in production unless it's for a download folder which helps us let them know this is not meant for production by any means.